pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -959,12 +1101,15 @@ func (c *Controller) AdditionalPodSubscriptions(
 	// Next, as an optimization, we will send all node-local endpoints
 	if nodeName := proxy.Metadata.NodeName; nodeName != "" {
 		for _, wl := range c.ambientIndex.All() {
-			if wl.Node == nodeName {
-				n := types.NamespacedName{Name: wl.ResourceName()}
-				if currentSubs.Contains(n) {
-					continue
+			switch addr := wl.Address.Type.(type) {
+			case *workloadapi.Address_Workload:
```

Comment:
let's add a case to explicitly ignore the service with the reason why so it doesn't look like it was just forgotten

Reply:
same as above, except this one I felt should be ignored since this is for node local - and Service's are not node aware so they are never returned here